client/js/components/newsletter-signup.js

```diff
@@ -0,0 +1,63 @@
+export default (el) => {
+  const emailInputEl = el.querySelector('input[type="email"]');
+  const checkboxEls = [...el.querySelectorAll('input[type="checkbox"]')];
+  const checkboxErrorMessage = document.createElement('p');
+  const emailErrorMessage = document.createElement('p');
+  const errorClasses = ['border-width-1', 'border-color-red', 'padding-left-s2', 'padding-right-s2', 'padding-top-s2', 'padding-bottom-s2', 'margin-bottom-s2', 'font-red'];
+  let isSubmitAttempted = false;
+
+  checkboxErrorMessage.classList.add(...errorClasses);
+  checkboxErrorMessage.innerHTML = 'Please select an option.';
+  emailErrorMessage.classList.add(...errorClasses);
+  emailErrorMessage.innerHTML = 'Please enter a valid email address.';
+
+  el.setAttribute('novalidate', '');
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    isSubmitAttempted = true;
+
+    if (!emailInputEl.validity.valid) {
+      el.classList.add('is-error');
+      el.append(emailErrorMessage);
```

Comment:
We could just use browser default error handling for this?

Reply:
We use browser default error handling, but progressively enhance it.